packages/react/src/ActionList/Item.tsx

```diff
@@ -294,7 +294,7 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
       'aria-disabled': disabled ? true : undefined,
       'data-inactive': inactive ? true : undefined,
       'data-loading': loading && !inactive ? true : undefined,
-      tabIndex: disabled || showInactiveIndicator ? undefined : 0,
+      tabIndex: (disabled && !inferredItemRole) || showInactiveIndicator ? undefined : 0,
```

Comment:
1. Should this be inferredItemRole or itemRole?
2. Why do we check for inferredItemRole being falsy/undefined?

----

Also, this statement has become very difficult to understand 😅 

Can we pull a variable out like `focusable` out of this object? so that we can expand this logic like:

```ts
let focusable = true // default
if (showInactiveIndicator) focusable = false // inactive elements are not focusable
else if (disabled && inferredItemRole === undefined) focusable = false // disabled items should get focus unless they have role=something

const menuItemProps = {
  tabIndex: focusable ? 0 : undefined
}
```

Reply:
> Should this check include role passed via props? (itemRole vs inferredItemRole)

I feel like we don't need to check `itemRole` since it doesn't determine if a `tabindex` should be added or not currently (I think!). I'm also wondering on custom implementations of `ActionList` where we wouldn't want to determine if the item receives an `tabindex` or not 🤔 

> Can we pull a variable out like focusable out of this object? so that we can expand this logic like:

Yup, this is great! I'll decouple it!